### ISSUE TITLE
Social login callback page sends the token twice

### ISSUE THREAD

**Issue by infomiho** [wasp-lang/wasp#1272]
Our social flow includes redirecting from e.g. Google to our frontend with an auth token in the query params. 

When our frontend page loads, it sends a request to our backend which then returns a token to the frontend.

After upgrading to React 18 and enabling their new `StrictMode`, each component is rendered twice in development. This means, our social auth callback page is rendered twice -> which sends the auth token twice to the backend.

This is an issue because the token is only valid **once** and the second request ends with an `500` error on our server which then logs out the user.

In a nutshell, our social auth flow is broken in development (it would still work in production builds).

**Comment by Martinsos** [comment]
Any ideas how to fix this?

**Comment by infomiho** [comment]
We need to redesign a bit how we send that request in the component, and it should be fine. Either we'll use an `AbortController` to cancel requests or we'll modify the query after the request is sent for the first time. 

I still have to investigate. 

https://stackoverflow.com/a/72238236
